gordo_components/model/factories/model_factories_utils.py

```diff
@@ -15,3 +16,36 @@ def hourglass_calc_dims(compression_factor, encoding_layers, n_features):
         round(n_features - (i * average_slope)) for i in range(1, encoding_layers + 1)
     ]
     return dims
+
+
+def _check_if_only_dim_or_func(dim, func):
```

Comment:
This PR isn't marked with `WIP` so I'll go ahead and say that for all these functions, we'll need type hints and numpy style docs for them.